src/risk_distributions/risk_distributions.py

```diff
@@ -114,328 +132,440 @@ def process(self, data: Union[np.ndarray, pd.Series], process_type: str,
+                raise ValueError(f"If passing weights as a dictionary, you "
+                                 f"must supply only keys {cls.distribution_map.keys()}.")
+            weights = {key: list(val) for key, val in weights.items()}
+            if len(set(len(val) for val in weights.values())) != 1:
+                raise ValueError("If passing weights as a dictionary, you "
+                                 "must specify the same number of weights for each distribution.")
+            weights = pd.DataFrame(weights)
+
+        params = {}
+        for name, dist in cls.distribution_map.items():
+            try:
+                param = parameters[name] if parameters else None
+                params[name] = dist.get_parameters(param, mean, sd)
+            except NonConvergenceError:
+                if weights[name].max() < 0.05:
+                    weights.loc[name, :] = 0
+                else:
+                    raise NonConvergenceError(f'Divergent {name} distribution has '
+                                              f'weights: {100 * weights[name]}%', name)
 
-        Parameters
-        ----------
-        weights :
-            A single-row dataframe of normalized distribution weights, each column representing
-            a different distribution type.
-        mean :
-            Mean value for a single distribution or series of values, each for a single distribution.
-        sd :
-            Standard deviation value for a single distribution or series of values, each for a single distribution.
+        # Rescale weights in case we floored any of them:
+        non_zero_rows = weights[weights.sum(axis=1) != 0]
+        weights.loc[non_zero_rows.index] = non_zero_rows.divide(non_zero_rows.sum(axis=1), axis=0)
 
-        Returns
-        -------
-            Rescaled weights and the subset of the distribution map corresponding to convergent distributions.
-        """
+        return weights, params
 
-        distributions = dict()
-        distribution_map = self.get_distribution_map()
-        distribution_types = list(set(distribution_map.keys()) & set(weights.columns))
-        weights = weights[distribution_types].iloc[0]
-        weights = weights/np.sum(weights)
-        for name in distribution_types:
-            try:
-                distributions[name] = distribution_map[name](mean=mean, sd=sd)
-            except NonConvergenceError as e:
-                if weights[e.dist] < 0.05:
-                    weights = weights.drop(e.dist)
-                else:
-                    raise NonConvergenceError(f'Divergent {key} distribution has weights: {100*weights[name]}%', name)
+    def pdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
+        single_val = isinstance(x, (float, int))
+        values_only = isinstance(x, np.ndarray)
 
-        return weights/np.sum(weights), distributions
+        x = pd.Series(x, index=self.weights.index)
```

Comment:
i think this will cause problems if x is a Series and doesn't have exactly the same index as weights (which seems distinctly likely)